### ISSUE TITLE
Preview Instant messages and Attachments in Chat Preview

### ISSUE THREAD

**Issue by aberenguel** [sepinf-inc/IPED#2291]
When we click on a image or video in References table, it is shown the file content Preview tab.

![image](https://github.com/user-attachments/assets/7f4aced1-755b-4ab3-9d0f-8e1415ab8be1)

Similarly, when we click on InstantMessage in SubItems table, it is shown the Metadata tab. The preview tab has no content.

![image](https://github.com/user-attachments/assets/6c6a513b-c6e3-4d3b-ad57-0c25c477473f)

It would be interesting if it could to navigate to the item position in chat preview. It would facilitate a lot some analysis.

**Comment by lfcnassif** [comment]
Right. I'm just not sure about changing the default metadata view, maybe the "go to chat position" context menu option could be added the related items panels.

**Comment by lfcnassif** [comment]
Question: To list those messages in subitems tab, the parent chat preview was selected on the main table and was displayed in the natural viewer, right? I don't see how just the message name, in the subitems tab, would help identifying a specific message as important so the user would want to view it in the chat context...

But if the user is analyzing instant messages on the main table, where each message detail can already be found in the table columns, displaying the message context in the chat preview makes sense to me. Just not sure how to implement it, because the chat preview and the instant message are different items and the current viewer API AFAIK just accepts 1 item at a time.

**Comment by aberenguel** [comment]
> Question: To list those messages in subitems tab, the parent chat preview was selected on the main table and was displayed in the natural viewer, right?

Yes.

> I don't see how just the message name, in the subitems tab, would help identifying a specific message as important so the user would want to view it in the chat context...

For messages (subitems) I agree with you, unless the message was bookmarked as important and it is colored in subitems table. But for images (references) it is great to see where are the images in chat classified as important (bookmaked as well).

> But if the user is analyzing instant messages on the main table, where each message detail can already be found in the table columns, displaying the message context in the chat preview makes sense to me. Just not sure how to implement it, because the chat preview and the instant message are different items and the current viewer API AFAIK just accepts 1 item at a time.

I also thought about that. I would have a workaround in MultiViewer like "if is a message, render its parent chat".

**Comment by lfcnassif** [comment]
> Just not sure how to implement it, because the chat preview and the instant message are different items and the current viewer API AFAIK just accepts 1 item at a time.

I just remembered the `ViewerController` class has the method below:
`public void loadFile(IStreamSource file, IStreamSource viewFile, String contentType, Set<String> highlightTerms)`

It is used, for example, for videos, for which the original item is passed to the first param, so Hex and Text viewers use the original video content, and the video preview/mosaic is passed as the second param, so it can be rendered in the Preview viewer.

If we decide to change current behavior, the instant message could be passed as the first param and its parent chat as the second one.
